cni: Benchmarks sporadically fail because we are not ending in the expected netns

We should have benchmarks consistently passing, but instead they fail sporadically because of CNI errors:
```
BenchmarkCreateStartStopDeletePodQemuHypervisorHyperstartAgentNetworkCNI-4    	--- FAIL: BenchmarkCreateStartStopDeletePodQemuHypervisorHyperstartAgentNetworkCNI-4
	api_test.go:1405: Could not delete pod: exit status 2: iptables v1.4.21: Couldn't load target `CNI-2e3a43835ef481fda21efc6f':No such file or directory
		
		Try `iptables -h' or 'iptables --help' for more information.
BenchmarkCreateStartStopDeletePodQemuHypervisorNoopAgentNetworkCNI-4          	--- FAIL: BenchmarkCreateStartStopDeletePodQemuHypervisorNoopAgentNetworkCNI-4
	api_test.go:1405: Could not delete pod: exit status 2: iptables v1.4.21: Couldn't load target `CNI-b0b4f04a417de990fd3ddeed':No such file or directory
		
		Try `iptables -h' or 'iptables --help' for more information.
```

The root cause is about our code being in the wrong netns and unable to see what it is supposed to delete. Still need to figure out the reason for this mixing between netns.

--- a/cnm.go
+++ b/cnm.go
@@ -181,7 +181,7 @@ func (n *cnm) run(networkNSPath string, cb func() error) error {
 		return fmt.Errorf("networkNSPath cannot be empty")
 	}
 
-	return safeDoNetNS(networkNSPath, func(_ ns.NetNS) error {
+	return doNetNS(networkNSPath, func(_ ns.NetNS) error {
 		return cb()
 	})
 }

--- a/network.go
+++ b/network.go
@@ -21,6 +21,7 @@ import (
 	"fmt"
 	"net"
 	"os"
+	"runtime"
 
 	"github.com/01org/ciao/ssntp/uuid"
 	"github.com/containernetworking/cni/pkg/ns"
@@ -302,24 +303,13 @@ func setNetNS(netNSPath string) error {
 	return n.Set()
 }
 
-// doNetNS makes some subsequent calls to a go routine and LockOSThread
-// in order to protect the current process from other thread switching
-// to different netns. Unless we have to make sure the thread ID has to
-// stay the same, this function should be used by default.
+// doNetNS is free from any call to a go routine, and it calls
+// into runtime.LockOSThread(), meaning it won't be executed in a
+// different thread than the one expected by the caller.
 func doNetNS(netNSPath string, cb func(ns.NetNS) error) error {
-	n, err := ns.GetNS(netNSPath)
-	if err != nil {
-		return err
-	}
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
 
-	return n.Do(cb)
-}
-
-// safeDoNetNS is free from any call to a go routine, meaning it will
-// not be executed in a different thread than the one expected by the
-// caller. This is used in case of CNM network, because we need to
-// make sure the process switched to the given netns has PID == TID.
-func safeDoNetNS(netNSPath string, cb func(ns.NetNS) error) error {
 	currentNS, err := ns.GetCurrentNS()
 	if err != nil {
 		return err

